src/Symfony/Component/Webhook/Client/RequestParser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Webhook\Client;
+
+use Symfony\Component\HttpFoundation\ChainRequestMatcher;
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestMatcher\IsJsonRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcher\MethodRequestMatcher;
+use Symfony\Component\HttpFoundation\RequestMatcherInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\RemoteEvent\Event;
+use Symfony\Component\Webhook\Exception\RejectWebhookException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 6.3
+ */
+class RequestParser extends AbstractRequestParser
+{
+    public function getSuccessfulResponse(): Response
+    {
+        return new Response('', 202);
+    }
+
+    public function getRejectedResponse(string $reason): Response
+    {
+        return new Response($reason, 406);
+    }
+
+    protected function getRequestMatcher(): RequestMatcherInterface
+    {
+        return new ChainRequestMatcher([
+            new MethodRequestMatcher('POST'),
+            new IsJsonRequestMatcher(),
+        ]);
+    }
+
+    protected function doParse(Request $request, string $secret): Event
+    {
+//$body = $this->bodyParser->parse($request, $secret);
+        $body = $request->toArray();
+
+//$this->headerParser->validate($request, $secret);
+        if (!$request->headers->has('x-id')
+            || !$request->headers->has('x-event')
+            || !$request->headers->has('x-signature')
+        ) {
+            throw new RejectWebhookException(406, 'Missing signature.');
```

Comment:
Just wondering: should these error messages be more verbose to contain details?

```
throw new RejectWebhookException(406, 'Missing signature. Include a signature in the "x-id", "x-event" or "x-signature" HTTP request headers');
```

I'm thinking in cases where you make a typo in the header name (e.g. `x-singature`) and maybe seeing the right header names can help you.

Reply:
Improved